raiden-ts/src/transfers/utils.ts

```diff
@@ -59,12 +59,50 @@ export function makeMessageId(): UInt<8> {
   return bigNumberify(Date.now()) as UInt<8>;
 }
 
+/**
+ * Return the transfer direction for a TransferState
+ *
+ * @param state - transfer to get direction from, or TransferId
+ * @returns Direction of transfer
+ */
+export function transferDirection(state: TransferState | { direction: Direction }): Direction {
+  if ('direction' in state) return state.direction;
+  const transfer = state.transfer[1];
+  return transfer.recipient === state.partner ? Direction.SENT : Direction.RECEIVED;
+}
+
+/**
+ * Get a unique key for a tranfer state or TransferId
+ *
+ * @param state - transfer to get key from, or TransferId
+ * @returns string containing a unique key for transfer
+ */
+export function transferKey(
```

Comment:
Just out of curiocity what was the problem that this key solves?

Reply:
Same as `channelKey`, hardcoding `secrethash` didn't account for duplications, in transfer's case, incoming and outgoing transfers could have the same secrethash (i.e. mediated transfers), and also secrethash only as a key didn't tell if it was an incoming our outgoing transfer, which was done at a time we only had outgoing transfers.
This _standardizes_ the "uniqueness" of key through this function, and if needed, we can update it in the future, to include other parameters for transfer uniqueness.